src/track_func.h

```diff
@@ -331,6 +331,28 @@ static inline TrackdirBits TrackBitsToTrackdirBits(TrackBits bits)
 	return (TrackdirBits)(bits * 0x101);
 }
 
+/**
+ * Checks whether a TrackBits has a given Track.
+ * @param tracks The track bits.
+ * @param track The track to check.
+ */
+static inline bool HasTrack(TrackBits tracks, Track track)
```

Comment:
There is already function named HasTrack rail_map.h, dunno how it even works currently but one of them has to be renamed imo.

Reply:
The function on rail_map.h takes a TileIndex and a track as arguments. As long as the type of a TrackBits is a UINT16, there should be no problem.
If it is necessary to change its name, just propose a name and I will do it.

Reply:
Well, it sure does not increase readability when you have figure out argument types and remember overloading rules. Also having overloaded functions doing different things is basically a time bomb. As for name there is pretty obvious name TrackBitsHasTrack. But since it's a codestyle thing you may want to consult @frosch123 or some other dev who's is actually able to merge the PR. I just mentioned it so they won't accidently miss it.

Reply:
Giving similar functions such as HasBit(param...) I wouldn't change the name of the HasTrack(param...). If the first parameter is a TileIndex, it will check if the tile has a given track. If the first parameter is a TrackBits, it will check if the TrackBits has a given track.